### PR TITLE
Use present_list() in document_symbol(), too

### PR DESCRIPTION
**Pull request by runiq**
Here's a patch for `text_document_document_symbol` on top of https://github.com/autozimu/LanguageClient-neovim/pull/1060 that uses `present_list()`.

The diff looks hairier than it actually is—`text_document_document_symbol()` now basically just reuses `Vec<SymbolInformation>` or computes a flat `Vec<DocumentSymbol>` via `walk_document_symbol()` and lets `present_list()` deal with displaying things.

This patch allows to remove the `FromLSP` trait and its implementations since they're not needed anymore. Also, `walk_document_symbol()` is defined in a single place now.

I unfortunately had to `clone()` the `DocumentSymbol` because of the change of the `name` field in child `DocumentSymbol`s (`[parent.name]::[child.name]`). If anyone knows of a better way to do this, I'm all ears. If this functionality is not needed (i.e. the document symbols can stay as they are, without the `[parent]::` prefix), that would simplify this code a lot.

One thing that would have helped me would be an `IntoIterator` implementation for `DocumentSymbol`, but that isn't available AFAICT. Guess I'll propose it over in lsp-types. (Edit: [Done](https://github.com/gluon-lang/lsp-types/issues/162).)

### CODE REVIEW

File: src/language_server_protocol.rs [hunk 1]
```diff
@@ -1420,111 +1420,46 @@ impl LanguageClient {
...
+                    let n = lsp_types::DocumentSymbol {
+                        name: name,
+                        ..ds.clone()
+                    };
 
-                        symbols
-                    }
-                    _ => Vec::new(),
-                };
+                    buffer.push(n);
```
Comment (martskins) [inline src/language_server_protocol.rs:1446]: Just a small nit to appease clippy, the `name: name` thing could be just `name`. And while we are at it, this whole block could just be:
`buffer.push(lsp_types::DocumentSymbol { name, ..ds.clone() });`

File: src/language_server_protocol.rs [hunk 2]
```diff
@@ -1420,111 +1420,46 @@ impl LanguageClient {
...
+                        ds.name.clone()
+                    };
 
-                        for ds in &nested {
-                            walk_document_symbol(&mut symbols, None, ds);
-                        }
+                    let n = lsp_types::DocumentSymbol {
+                        name: name,
+                        ..ds.clone()
+                    };
```
Comment (martskins) [inline src/language_server_protocol.rs]: Just a small nit to appease clippy, the `name: name` thing could be just `name`. And while we are at it, this whole block could just be:
`buffer.push(lsp_types::DocumentSymbol { name, ..ds.clone() });`

**Comment by martskins** [comment]
Left just one comment about a minor linting issue to address, other than that it looks good! Thank you!

**Comment by martskins** [comment]
> One thing that would have helped me would be an IntoIterator implementation for DocumentSymbol

About that, what were you planning on iterating over? Cause the children of a `DocumentSymbol` are an `Option<Vec<DocumentSymbol>>`, so you could just `.iter()` that if that's what you wanted to iterate over.

**Comment by runiq** [comment]
> Left just one comment about a minor linting issue to address

Addressed!

**Pull request merged**